src/test/java/UploadReceiptServletTest.java

```diff
@@ -394,6 +381,42 @@ public void doPostThrowsIfReceiptAnalysisFails() throws IOException {
     verify(blobstoreService).delete(BLOB_KEY);
   }
 
+  @Test
+  public void doPostRoundPrice() throws IOException {
+    helper.setEnvIsLoggedIn(true);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    createMockBlob(request, VALID_CONTENT_TYPE, VALID_FILENAME, IMAGE_SIZE_1MB);
+
+    double price = 17.236;
+    double roundedPrice = 17.24;
+    stubRequestBody(request, LABEL, STORE, price, PAST_TIMESTAMP);
+    stubUrlComponents(
+        request, LIVE_SERVER_SCHEME, LIVE_SERVER_NAME, LIVE_SERVER_PORT, LIVE_SERVER_CONTEXT_PATH);
+
+    // Mock receipt analysis.
+    mockStatic(ReceiptAnalysis.class);
+    when(ReceiptAnalysis.serveImageText(new URL(LIVE_SERVER_ABSOLUTE_URL)))
+        .thenReturn(ANALYSIS_RESULTS);
+
+    servlet.doPost(request, response);
+
+    Query query = new Query("Receipt");
+    PreparedQuery results = datastore.prepare(query);
+    Entity receipt = results.asSingleEntity();
+
+    Assert.assertEquals(receipt.getProperty("imageUrl"), IMAGE_URL);
+    Assert.assertEquals(receipt.getProperty("price"), roundedPrice);
+    Assert.assertEquals(receipt.getProperty("store"), STORE);
+    Assert.assertEquals(receipt.getProperty("rawText"), RAW_TEXT);
+    Assert.assertEquals(receipt.getProperty("blobKey"), BLOB_KEY);
+    Assert.assertEquals(receipt.getProperty("timestamp"), PAST_TIMESTAMP);
+    Assert.assertEquals(receipt.getProperty("label"), LABEL);
```

Comment:
I would probably only assert on the price since the others are redundant with the assertions in doPostUploadsReceiptToDatastoreLiveServer. Generally we don't want to assert more than is required to verify a behavior.